File: app/src/main/java/com/vn/ezlearn/config/AppConfig.java
package com.vn.ezlearn.config;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by manhi on 8/6/2016.
 */

public class AppConfig {
    private static final String Pref = "Pref";

    private static final String KEY_IS_SHOW_INTRO = "key_is_shcow_intro";
    private static final String KEY_TOKEN = "key_token";
    private static final String KEY_NAME = "key_name";
    private static final String KEY_EMAIL = "key_email";

    private SharedPreferences sharedPreferences;
    private static AppConfig appConfig;

    public static AppConfig getInstance(Context context) {
        if (appConfig == null) {
            appConfig = new AppConfig(context);
        }
        return appConfig;
    }

    private AppConfig(Context context) {
        this.sharedPreferences = context.getSharedPreferences(Pref, Context.MODE_PRIVATE);
    }

    public void setIsShowIntro(boolean isShow) {
        sharedPreferences.edit().putBoolean(KEY_IS_SHOW_INTRO, isShow).apply();
    }

    public boolean isShowIntro() {
        return sharedPreferences.getBoolean(KEY_IS_SHOW_INTRO, false);
    }

    public void setToken(String token) {
        sharedPreferences.edit().putString(KEY_TOKEN, token).apply();
    }

    public String getToken() {
        return sharedPreferences.getString(KEY_TOKEN, "");
    }

    public void setEmail(String email) {
        sharedPreferences.edit().putString(KEY_EMAIL, email).apply();
    }

    public String getEmail() {
        return sharedPreferences.getString(KEY_EMAIL, "");
    }

    public void setName(String name) {
        sharedPreferences.edit().putString(KEY_NAME, name).apply();
    }

    public String getName() {
        return sharedPreferences.getString(KEY_NAME, "");
    }
}

File: app/src/main/java/com/vn/ezlearn/config/AppConfig.kt
package com.vn.ezlearn.config

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by manhi on 8/6/2016.
 */

class AppConfig private constructor(context: Context) {

    private val sharedPreferences: SharedPreferences

    var isShowIntro: Boolean
        get() = sharedPreferences.getBoolean(KEY_IS_SHOW_INTRO, false)
        set(isShow) = sharedPreferences.edit().putBoolean(KEY_IS_SHOW_INTRO, isShow).apply()

    var isSelectLevel: Boolean
        get() = sharedPreferences.getBoolean(KEY_IS_SELECT_LEVEL, false)
        set(isSelectLevel) = sharedPreferences.edit().putBoolean(KEY_IS_SELECT_LEVEL, isSelectLevel).apply()

    var studyLevelID: Int
        get() = sharedPreferences.getInt(KEY_ID_STUDY_LEVEL, 0)
        set(id) = sharedPreferences.edit().putInt(KEY_ID_STUDY_LEVEL, id).apply()

    var degreeID: Int
        get() = sharedPreferences.getInt(KEY_ID_DEGREE, 0)
        set(id) = sharedPreferences.edit().putInt(KEY_ID_DEGREE, id).apply()

    init {
        this.sharedPreferences = context.getSharedPreferences(Pref, Context.MODE_PRIVATE)
    }

    fun clearData() {
        sharedPreferences.edit().clear().apply()
    }

    companion object {
        private val Pref = "Pref"

        private val KEY_IS_SHOW_INTRO = "key_is_show_intro"
        private val KEY_IS_SELECT_LEVEL = "key_is_select_level"
        private val KEY_ID_STUDY_LEVEL = "key_id_study_level"
        private val KEY_ID_DEGREE = "key_id_degree"
        private var appConfig: AppConfig? = null

        fun getInstance(context: Context): AppConfig {
            if (appConfig == null) {
                appConfig = AppConfig(context)
            }
            return appConfig as AppConfig
        }
    }
}
